### DISCUSSION TITLE
Смена языка интеграции

### DISCUSSION THREAD

**Discussion by YujiTFD** [IATkachenko/HA-YandexWeather#29] [Ideas]
### В чем проблема?

Здравствуйте.

Помогите, пожалуйста, понять, как изменить язык, используемый самой интеграцией.
По старой привычке в системе использую английскую локаль, всё устраивает, но погоду хочется видеть на русском языке.
Локаль менял, перезагружал, переустанавливал - всё едино, только английские именования. В папке интеграции вижу файл **const.py**, неужели язык меняется только прямым изменением констант вручную?

### При использовании какой версии интеграции возникает проблема?

0.8.1

### Какая версия Home Assistant используется?

2022.4.6

### Какой способ установки Home Assistant используется?

Home Assistant OS

**Comment by IATkachenko** [comment]
Добрый день.

Я правильно понимаю что речь идет про этот блок:
![изображение](https://user-images.githubusercontent.com/17473830/166102482-035355dd-199a-4b33-8865-997b24a9982d.png)
и связанные с ним вещи (вроде карточек с показаниями сенсоров)?

В общем случае у вас ничего не получится: HA сам занимается переводом интеграций на используемую в инсталляции локаль. Те если в интеграции есть перевод на русский, а локаль инсталляции английская, все тексты всеравно будут на английском. Те то что вы видите в `const.py` напрямую в UI не попадает, а в любом случае преобразуется с учетом выбранной локали. По умолчанию все на английском. Если в интеграции есть перевод на локаль инсталляции -- используется перевод.

Но да, русификации добавить надо бы...

**Comment by YujiTFD** [comment]
Да, вы верно меня поняли.
Я задал вопрос потому, что в https://github.com/IATkachenko/HA-YandexWeather/issues/27 увидел русскую локаль, заинтересовался и начал экспериментировать.
Про привязку языка интеграции к локали инсталляции - плохо: максимум информации по большинству вопросов в интернете на английском языке, и за много лет я уже очень устал переводить в голове уже въевшиеся в голову термины, определения и фразы на другой язык и обратно :(

**Comment by IATkachenko** [comment]
Можно, конечно исхитриться и частично все сработает: в `sensor.py` есть массив сенсоров. У каждого есть поле `name`,
например:
```python
    SensorEntityDescription(
        key=ATTR_API_HUMIDITY,
        name="Humidity",
        native_unit_of_measurement=PERCENTAGE,
        device_class=SensorDeviceClass.HUMIDITY,
        state_class=SensorStateClass.MEASUREMENT,
        entity_registry_enabled_default=False,
    ),
```
и это именно то, что вам нужно.
На уровне интеграции изменять я не готов (это сломает кучу всего в действующих инсталляциях и я не считаю правильным что базовое имя, с которого идет перевод на другие языки, -- на русском).

С атрибутами `climate` такой трюк не пройдет. 

Еще вы можете сделать `translations/sensor.en.json` и `translations/climate.en.json` (именно en), где для каждого имени сенсора и атрибута прописать русский перевод.
в виде 
```json
{
  "state": {
    "yandex_weather__status": {
      "Humidity": "Влажность"
    }
  }    
}
```
Тогда HA подхвати это для вашей английской локали и у вас все будет русифицировано.
Но это, опять же, на уровне релиза применено не будет, по понятным причинам.

Так что, видимо, wontfix, а обсуждение можем продолжить в дискуссиях.

**Comment by YujiTFD** [comment]
И снова здравствуйте.

Я прошу прощения за назойливость, но конкретно в этом кодинге я почти полный ноль, "иду по приборам".
`translations/climate.en.json` ничего не дал ни сразу, ни после перезагрузки. Я что-то недоделал?

```
{
  "state": {
    "yandex_weather__status": {
      "Humidity": "Влажность"
      "Pressure mmHg": "Давление мм рт.ст."
    }
  }    
}

```

**Reply by IATkachenko** [reply]
@YujiTFD, все в порядке. Я для этого и переместил в обсуждения, чтобы мы могли прожолжить.

У вас json не валидный. Запятой не хватает:
```json
{
  "state": {
    "yandex_weather__status": {
      "Humidity": "Влажность",
      "Pressure mmHg": "Давление мм рт.ст."
    }
  }    
}
```

**Reply by IATkachenko** [reply]
Куда-то не в ту степь я вас направил: это -- перевод именно для состояний (значений).
Простого способа перевести именно  названия HA не предлагет. Но это можно сделать через UI, просто изменив отображемое имя сенсора.
![изображение](https://user-images.githubusercontent.com/17473830/167259917-568b3cbd-e315-4b92-b8df-614c3b660dac.png)

**Reply by YujiTFD** [reply]
Хорошо, с описанием разобрались, не проблема переименовать, спасибо.
А что делать с состояниями? "mmHg", "partlycloudly", "5 m/s" вместо "5 м/c" - это как поправить?

**Reply by IATkachenko** [reply]
`partlycloudly` -- это как раз в переводе состояний правится. Я почти доделал русские состояния, которые вы сможете использовать.
Остальное (единицы измерения) -- в вашем случае, боюсь, никак -- это уже явно внутри языковой системы HA.

**Reply by nikc547** [reply]
Я правильно понимаю, что перевод только для карточек?
Если создавать свои сенсоры на основе сенсора weather.yandex_weather - то там необработанные данные?
В таком случае, в template sensor надо обработку делать и привязывать русское значение к английскому по примеру:
        value_template: >-
          {% if is_state('condition', 'clear') %}
            Ясно
          {% elif is_state('condition', 'cloudy') %}
            Облачно
          {% endif %}

**Reply by IATkachenko** [reply]
@nikc547, добрый день.
Да, все верно.
А каких сенсоров вам не хватает? Откройте feature request -- я посмотрю что можно добавить. Может они будут не только вам полезны ;)

**Reply by nikc547** [reply]
Добрый вечер!
Штатно датчиков вполне хватает.
У меня они не завелись, но времени не было разбираться.
Пару датчиков создал из атрибутов для телеграмма.
Завтра разберусь, если вск же не заработают - буду issue создавать

**Reply by nikc547** [reply]
> Завтра разберусь, если вск же не заработают - буду issue создавать

Все работает, я не терпеливый))
Для бота телеграмм использовал сенсор weather.yandex_weather, который отдавал state на английском.
Пришлось расписывать значения на русский. Спасибо за файлик с переводами - очень помог со списком отдаваемых значений.
Если кому нужен - ниже код сенсора:
  - platform: template
    sensors:
      weather_condition_now:
        friendly_name: "Погода "
        value_template: >-
          {% if is_state('weather.yandex_weather', 'clear-night') %} 
            Ясно
          {% elif is_state('weather.yandex_weather', 'cloudy') %} 
            Облачно
          {% elif is_state('weather.yandex_weather', 'exceptional') %} 
            Предупреждение
          {% elif is_state('weather.yandex_weather', 'fog') %} 
            Туман
          {% elif is_state('weather.yandex_weather', 'hail') %} 
            Град
          {% elif is_state('weather.yandex_weather', 'lightning') %} 
            Молния
          {% elif is_state('weather.yandex_weather', 'lightning-rainy') %} 
            Молния, дождь
          {% elif is_state('weather.yandex_weather', 'partlycloudy') %} 
            Переменная облачность
          {% elif is_state('weather.yandex_weather', 'pouring') %} 
            Ливень
          {% elif is_state('weather.yandex_weather', 'rainy') %} 
            Дождь
          {% elif is_state('weather.yandex_weather', 'snowy') %} 
            Снег
          {% elif is_state('weather.yandex_weather', 'snowy-rainy') %} 
            Снег с дождем
          {% elif is_state('weather.yandex_weather', 'sunny') %} 
            Ясно
          {% elif is_state('weather.yandex_weather', 'windy') %} 
            Ветрено
          {% elif is_state('weather.yandex_weather', 'windy-variant') %} 
            Ветрено
          {% endif %}

**Comment by IATkachenko** [comment]
В [v1.2.0](https://github.com/IATkachenko/HA-YandexWeather/releases/tag/v1.2.0) появилась возможность выбрать язык, на котором сенсор `sensor.yandex_weather_condition_yandex` будет отдавать свое состояние.
